### ISSUE TITLE
Table: sortable header cell does not have sufficient contrast on hover

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#32301]
![image](https://github.com/user-attachments/assets/cf2bb74e-9a05-4e39-affa-c9447fc11abe)

Can be repro-d on docsite
